lib/puppet/transaction/resource_harness.rb

```diff
@@ -92,6 +93,25 @@ def perform_changes(resource, context)
     end
 
     capture_audit_events(resource, context)
+    persist_system_values(resource, context)
+  end
+
+  # We persist the last known values for the properties of a resource after resource
+  # application.
+  # @param [Puppet::Type] resource resource whose values we are to persist.
+  # @param [ResourceApplicationContent] context the application context to operate on.
+  def persist_system_values(resource, context)
```

Comment:
No, it stores the "shortened" version, depending on the provider property this can be different, for file its the md5.

> Also - with respect to files... if the upstream source is a puppet:// style URI -- and the backing contents of that file change, is that classified as a corrective_change or is it expected?

So this all translates to content property actions. In this case its much like setting the content field, so it won't be a corrective_change, its intended.